src/test/java/io/phasetwo/service/resource/OrganizationResourceTest.java

```diff
@@ -1,71 +1,267 @@
+    try {
+      RealmModel realm = session.realms().getRealmByName("master");
+      UserModel user = session.users().getUserByUsername(realm, "admin");
+      OrganizationProvider provider = session.getProvider(OrganizationProvider.class);
+      Stream<OrganizationModel> orgs = provider.searchForOrganizationStream(realm, ImmutableMap.of("name", "FOO"), 0, 50, Optional.empty());
+      OrganizationModel org = orgs.findFirst().get();
+      assertNotNull(org);
+      assertThat(org.getId(), is(id));
+
+      orgs = provider.searchForOrganizationStream(realm, ImmutableMap.of("name", "fO"), 0, 50, Optional.empty());
+      org = orgs.findFirst().get();
+      assertNotNull(org);
+      assertThat(org.getId(), is(id));
+
+      orgs = provider.searchForOrganizationStream(realm, ImmutableMap.of("name", "Oo cORp"), 0, 50, Optional.empty());
+      org = orgs.findFirst().get();
+      assertNotNull(org);
+      assertThat(org.getId(), is(id));
+
+      orgs = provider.searchForOrganizationStream(realm, ImmutableMap.of("name", "foo"), 0, 50, Optional.empty());
+      org = orgs.findFirst().get();
+      assertNotNull(org);
+      assertThat(org.getId(), is(id));
+      session.getTransactionManager().commit();
+    } finally {
+      session.close();
+    }
+
+
+    // check no crossover realm domains
+    session = factory.create();
+    session.getTransactionManager().begin();
+    try {
+      RealmModel realm = session.realms().getRealmByName("master");
+      OrganizationProvider provider = session.getProvider(OrganizationProvider.class);
+      Stream<OrganizationModel> orgs = provider.getOrganizationsStreamForDomain(realm, "foo.com", false);
+      assertThat(orgs.count(), is(1l));
+      orgs = provider.getOrganizationsStreamForDomain(realm, "foo.com", false);
+      assertThat(orgs.collect(MoreCollectors.onlyElement()).getName(), is("foo"));
+
+      realm = session.realms().getRealmByName("test");
+      orgs = provider.getOrganizationsStreamForDomain(realm, "foo.com", false);
+      assertThat(orgs.count(), is(1l));
+      orgs = provider.getOrganizationsStreamForDomain(realm, "foo.com", false);
+      assertThat(orgs.collect(MoreCollectors.onlyElement()).getName(), is("bar"));
+
+      session.getTransactionManager().commit();
+    } finally {
+      session.close();
+    }
+
+    // remove
+    session = factory.create();
+    session.getTransactionManager().begin();
+    try {
+      OrganizationProvider provider = session.getProvider(OrganizationProvider.class);
+
+      RealmModel realm = session.realms().getRealmByName("master");
+      boolean removed = provider.removeOrganization(realm, id);
+      assertTrue(removed);
+
+      realm = session.realms().getRealmByName("test");
+      removed = provider.removeOrganization(realm, barid);
+      assertTrue(removed);
+
+      session.getTransactionManager().commit();
+    } finally {
+      session.close();
+    }
+  }
+  */
+  /////////////////////////////////////
+
+  /*
```

Comment:
We need to have all of these tests uncommented and working

Reply:
I'll work on it and make incremental commits to avoid a large, final commit with all tests done. This way, I can keep track of the progress and address any issues as they arise.